modules/Preprocessing.py

```diff
@@ -89,17 +107,21 @@ def get_tweet_data(tweets: List[Dict[str, str]]):
         if (tweet_text != None and any(symbol in tweet_text for symbol in ['#', '@'])):
             for word in tweet_text.split(' '):
                 if word.startswith('#'):
+                    # clean hashtag
+                    clean_word = re.sub(_REGEX_CHAR_MATCHER_TWEETS, "", word)
                     # add to local hashtags
-                    tweet['hashtags'].append(word)
+                    tweet['hashtags'].append(clean_word)
                     # add to overall hashtags
-                    hashtag_stats[word.lower()] = hashtag_stats.get(word.lower(), 0) + 1
+                    hashtag_stats[clean_word.lower()] = hashtag_stats.get(clean_word.lower(), 0) + 1
                     # remove hashtag
                     tweet_text = tweet_text.replace(word, '')
                 if word.startswith('@'):
+                    # clean mention
+                    clean_word = re.sub(_REGEX_CHAR_MATCHER_TWEETS, "", word)
                     # add to local hashtags
-                    tweet['mentions'].append(word)
+                    tweet['mentions'].append(clean_word)
                     # add to overall hashtags
-                    mention_stats[word.lower()] = mention_stats.get(word.lower(), 0) + 1
+                    mention_stats[clean_word.lower()] = mention_stats.get(clean_word.lower(), 0) + 1
                     # remove mention
                     tweet_text = tweet_text.replace(word, '')
```

Comment:
Would love suggestions on how to handle this repeat code. I know there's an answer to it, maybe even without an external function, but it eludes me right now.

Reply:
No solution.